.github/workflows/coverage.yaml

```diff
@@ -149,13 +151,32 @@ jobs:
 
           # fill new tests
           # using `|| true` here because if no tests found, pyspec fill returns error code
+          FOUND_TEST=false
           mkdir -p fixtures/state_tests
           mkdir -p fixtures/eof_tests
-          echo "$files" | while read line; do
+
+          # Use a while loop with a here-string to avoid subshell issues
+          while IFS= read -r line; do
             file=$(echo "$line" | cut -c 3-)
-            fill $file --until=Cancun --evm-bin evmone-t8n || true >> filloutput.log 2>&1
-            (fill $file --fork=CancunEIP7692 --evm-bin evmone-t8n -k eof_test || true) > >(tee -a filloutput.log filloutputEOF.log) 2>&1
-          done
+            if grep -q "def test_" "$file"; then
+              FOUND_TEST=true
+              fill "$file" --until=Cancun --evm-bin evmone-t8n || true >> filloutput.log 2>&1
+              (fill "$file" --fork=CancunEIP7692 --evm-bin evmone-t8n -k eof_test || true) > >(tee -a filloutput.log filloutputEOF.log) 2>&1
```

Comment:
I think if we start adding hard-coded stuff here it will become very difficult to maintain this file.

We should declare an array of fill parameter combinations to use somewhere before this.

Reply:
I only want to run retesteth coverage via t8n for some time to be sure that t8n and evmone test runners cause the same code coverage. (Because all the time, teams pointed out that t8n is not as reliable as native test runners)

Reply:
I simplified the test logic now:
- the tests are not detected by script, if no tests it just fails (but it should be a rare case when we commit into test folder without touching the tests)
- the files are detected once and we can clearly see what .py files are executed by the script